Neos.Neos/Classes/ViewHelpers/Link/NodeViewHelper.php

```diff
@@ -317,30 +311,33 @@ public function render(): string
             }
         }
 
-        $uriBuilder = new UriBuilder();
-        $uriBuilder->setRequest($this->controllerContext->getRequest()->getMainRequest());
-        $uriBuilder->setFormat($this->arguments['format'])
-            ->setCreateAbsoluteUri($this->arguments['absolute'])
-            ->setArguments($this->arguments['arguments'])
-            ->setSection($this->arguments['section'])
-            ->setAddQueryString($this->arguments['addQueryString'])
-            ->setArgumentsToBeExcludedFromQueryString($this->arguments['argumentsToBeExcludedFromQueryString']);
+        $nodeUriBuilder = $this->nodeUriBuilderFactory->forRequest($this->controllerContext->getRequest()->getHttpRequest());
+
+        // todo ->setAddQueryString($this->arguments['addQueryString'])
+        // todo ->setArgumentsToBeExcludedFromQueryString($this->arguments['argumentsToBeExcludedFromQueryString']);
```

Comment:
what about those?

Reply:
Concerning the todos in the fluid view helpers and fusion impl. this change REMOVES certain features as discussed with martin as they should have been actually deprecated (we missed that):

lets discuss this in this space i just opened: https://github.com/neos/neos-development-collection/issues/5076

once we have a conclusion we can proceed. Lucky case would be to just remove it as proposed.

Reply:
pr to remove them which would allow us to move forward here: https://github.com/neos/neos-development-collection/pull/5079